frr-10.1.yaml

```diff
@@ -0,0 +1,111 @@
+package:
+  name: frr-10.1
+  version: 10.1.1
+  epoch: 0
+  description: The FRRouting Protocol Suite
+  copyright:
+    - license: GPL-2.0-only
+  dependencies:
+    provides:
+      - frr=${{package.full-version}}
+    runtime:
+      - python-3.11
+
+environment:
+  contents:
+    packages:
+      - autoconf
+      - automake
+      - bison
+      - build-base
+      - busybox
+      - elfutils-dev
+      - flex
+      - json-c-dev
+      - libcap-dev
+      - libtool
+      - libyang-dev
+      - linenoise-dev
```

Comment:
Is this needed? It isn't in the frrrouting/frr image

Reply:
these are build time dependencies and very much required for the image. The upstream image is based on alpine linux and their APKBUILD source is following https://github.com/FRRouting/frr/blob/960462aadedbc921a80c20c231cfcf6e94b5a98b/alpine/APKBUILD.in#L10-L22 

When I remove it and build it again then I run into the following:
```bash
2024/11/04 19:20:14 WARN /usr/lib/gcc/x86_64-pc-linux-gnu/14/../../../../x86_64-pc-linux-gnu/bin/ld: cannot find -llinenoise: No such file or directory uses=autoconf/make
```
and eventually it errors out.